src/vs/workbench/contrib/chat/common/promptSyntax/service/promptsService.ts

```diff
@@ -169,6 +172,7 @@ export class PromptsService extends Disposable implements IPromptsService {
 		});
 	}
 
+	@logTime()
```

Comment:
This pattern just seems a bit weird, that a decorator in base relies on the object exposing an ILogService publicly?

As long as TS gives us the proper errors, I'm ok with it but I wonder if you could do something like pass `this.logger` into `@logTime(this.logger)` or would that not be allowed?

Reply:
I guess I'm also worried about confusion from having two `ILogger` interfaces floating around?

Reply:
Yeah I think the name might be confusing, will try to find a better name for it. I was also was thinking about the `@logTime(this.logger)`, let me try to make it work 🤔